service/history/workflow/workflow_task_state_machine.go

```diff
@@ -446,12 +442,8 @@ func (m *workflowTaskStateMachine) AddWorkflowTaskCompletedEvent(
 	m.beforeAddWorkflowTaskCompletedEvent()
 	if workflowTask.Attempt > 1 {
 		// Create corresponding WorkflowTaskSchedule and WorkflowTaskStarted events for workflow tasks we have been retrying
-		taskQueue := &taskqueuepb.TaskQueue{
-			Name: m.ms.executionInfo.TaskQueue,
-			Kind: enumspb.TASK_QUEUE_KIND_NORMAL,
-		}
 		scheduledEvent := m.ms.hBuilder.AddWorkflowTaskScheduledEvent(
-			taskQueue,
+			m.ms.TaskQueue(),
```

Comment:
just to confirm: we clear stickiness on workflow task failure, so this method will always return a normal task queue here?

Reply:
It should. But even if it doesn't, it is better to log real task queue kind in the history rather than blindly overwrite it with `NORMAL`.